### PR TITLE
Add snapshot example in CSI doc

### PR DESCRIPTION
**Pull request by adisky**
Added Snapshot Example in Cinder CSI Docs

**What this PR does / why we need it**:
Updates  CSI Documentation for Snapshot Feature

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:
`NONE`

```release-note
```

### CODE REVIEW

File: docs/using-cinder-csi-plugin.md [hunk 1]
```diff
@@ -141,6 +141,41 @@ Currently, driver supports only one topology key: `topology.cinder.csi.openstack
 
 Note: `allowedTopologies` can be specified in storage class to restrict the topology of provisioned volumes to specific zones and should be used as replacement of `availability` parameter.
 
+### Example Snapshot Create and Restore
+
+Following prerequisite needed for volume snapshot feature to work.
+
+1. Enable `-feature-gates="VolumeSnapshotDataSource=true` in kube-apiserver
```
Comment (lingxiankong) [inline docs/using-cinder-csi-plugin.md]: quote missing

File: docs/using-cinder-csi-plugin.md [hunk 2]
```diff
@@ -141,6 +141,41 @@ Currently, driver supports only one topology key: `topology.cinder.csi.openstack
...
+
+For Snapshot Creation and Volume Restore, please follow  below steps:
+
+* Create Storage Class, Snapshot Class and PVC    
+```
+$ kubectl -f examples/cinder-csi-plugin/example.yaml create
+```     
+* Verify that pvc is bounded
+``` 
+$ kubectl describe pvc
```
Comment (lingxiankong) [inline docs/using-cinder-csi-plugin.md]: `kubectl describe pvc <pvc-name>`

File: docs/using-cinder-csi-plugin.md [hunk 3]
```diff
@@ -141,6 +141,41 @@ Currently, driver supports only one topology key: `topology.cinder.csi.openstack
...
+```       
+* Verify that snapshot is created    
+```
+$ kubectl get volumesnapshot 
+$ kubectl get volumesnapshotcontent
+```   
+* Restore volume from snapshot    
+```
+$ kubectl -f examples/cinder-csi-plugin/snapshotrestore.yaml create
+```
```
Comment (ramineni) [inline docs/using-cinder-csi-plugin.md]: could we add verify step here as well , the new volume is created with snapshot specified?

Comment (adisky) [reply docs/using-cinder-csi-plugin.md]: done

File: docs/using-cinder-csi-plugin.md (hunk 1 above)
Comment (adisky) [reply docs/using-cinder-csi-plugin.md]: done

File: docs/using-cinder-csi-plugin.md (hunk 2 above)
Comment (adisky) [reply docs/using-cinder-csi-plugin.md]: done

File: docs/using-cinder-csi-plugin.md [hunk 4]
```diff
@@ -141,6 +141,41 @@ Currently, driver supports only one topology key: `topology.cinder.csi.openstack
...
 
+### Example Snapshot Create and Restore
+
+Following prerequisite needed for volume snapshot feature to work.
+
+1. Enable `-feature-gates="VolumeSnapshotDataSource=true` in kube-apiserver
+2. Make Sure, your csi deployment contains external-snapshotter sidecar container, external-snapshotter sidecar container will create three crd's for snapshot management VolumeSnapshot,VolumeSnapshotContent, and VolumeSnapshotClass.       
+To deploy cinder csi driver and all required sidecars run
+```
+kubectl -f manifests/cinder-csi-plugin create
```
[the remark this answers is not in the document]
Comment (adisky) [reply docs/using-cinder-csi-plugin.md]: done

File: docs/using-cinder-csi-plugin.md (hunk 3 above)
Comment (adisky) [reply docs/using-cinder-csi-plugin.md]: done

**Pull request merged**